### DISCUSSION TITLE
TypeScript is giving me error TS2574: A rest element type must be an array type.

### DISCUSSION THREAD

**Discussion by zenhuzaini** [unifiedjs/unified#179] [Q&A] [answered]
Hi, when I run tsc tsconfig.json, I have an error TS2574 in unified module.
```node_modules/unified/index.d.ts:590:47 - error TS2574: A rest element type must be an array type.```
![image](https://user-images.githubusercontent.com/8146529/143407726-888765d7-2b15-46f4-bda9-0b47c6e1b4f0.png)

here is what in tsconfig.json
```
{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "module": "esnext",
    "lib": ["es2015", "es2017", "dom", "WebWorker.ImportScripts"],
    "target": "es5",
    "baseUrl": "src",
    "allowSyntheticDefaultImports": true,
    "noEmit": true,
    "removeComments": false,
    "downlevelIteration": true
  },
  "exclude": ["node_modules"],
  "include": ["src/**/*", "scripts/**/*.ts"]
}
```
recently for @types/react I am using the version ```17.0.5``` . I have tried to upgrade and downgrade the version but I still have the same error. and for unified, I have version ```^10.0.0```

**Comment by wooorm** [comment, marked as answer]
- What TS version?
- Are you writing your own plugins?

**Reply by zenhuzaini** [reply]
hi it was v3. after upgrading to v4 the error is gone. thanks :)
